adapters/pixfuture/pixfuture.go

```diff
@@ -0,0 +1,133 @@
+package pixfuture
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/iterutil"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+	headers := http.Header{}
+	headers.Set("Content-Type", "application/json")
+	headers.Set("Accept", "application/json")
+
+	for imp := range iterutil.SlicePointerValues(request.Imp) {
+		var bidderExt struct {
```

Comment:
Use the defined type instead of anonymous struct and add validation:

```go
var bidderExt adapters.ExtImpBidder
var pixfutureExt openrtb_ext.ImpExtPixfuture

if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err \!= nil {
    errs = append(errs, err)
    continue
}
if err := jsonutil.Unmarshal(bidderExt.Bidder, &pixfutureExt); err \!= nil {
    errs = append(errs, err)
    continue
}
if pixfutureExt.PixID == "" {
    errs = append(errs, &errortypes.BadInput{Message: "Missing required parameter pix_id"})
    continue
}
```

This ensures consistency with the defined schema type and adds runtime validation.